src/nft/utils/txRoute/combineItemsWithTxRoute.ts

```diff
@@ -31,14 +41,44 @@ const isAveragePriceOfPooledAssets = (
   return !isPriceDiff(calcAvgGroupPoolPrice(asset, numberOfAssetsInPool), expectedPrice)
 }
 
-const isAveragedPrice = (
+const isErc1155TokenCorrectSum = (item: UpdatedGenieAsset, items: UpdatedGenieAsset[], txRoute?: RoutingItem[]) => {
+  if (!txRoute) return false
+
+  const wishItems = items.filter(
+    (wishItem) =>
+      item.address === wishItem.address && item.tokenId === wishItem.tokenId && itemHasRoute(wishItem, txRoute)
+  )
+  const wishPrice = wishItems.reduce((acc, cur) => {
+    const newPriceInfo = cur.updatedPriceInfo ?? cur.priceInfo
+    const sum = acc.add(BigNumber.from(newPriceInfo.basePrice))
+    return sum
+  }, BigNumber.from(0))
+
+  const routeItems = wishItems.map((wishItem) => getRouteForItem(wishItem, txRoute))
+  const routePrice = routeItems.reduce((acc, cur) => {
+    if (cur && 'priceInfo' in cur.assetOut) {
+      const sum = acc.add(BigNumber.from(cur.assetOut.priceInfo.basePrice))
+      return sum
+    } else {
+      return acc
+    }
+  }, BigNumber.from(0))
+
+  return wishPrice.eq(routePrice)
+}
+
+const isCorrectCalculatedPrice = (
   item: UpdatedGenieAsset,
   items: UpdatedGenieAsset[],
   route: RoutingItem,
   txRoute?: RoutingItem[]
 ): boolean => {
   if (!(route && 'priceInfo' in route.assetOut)) return false
 
+  if (item.tokenType === TokenType.ERC1155) {
+    return isErc1155TokenCorrectSum(item, items, txRoute)
+  }
```

Comment:
I'm currently unable to buy an 1155 that is part of a liquidity pool and always get the error: `Error: cannot estimate gas; transaction may fail or may require manual gas limit`